src/pyhf/pdf.py

```diff
@@ -217,11 +217,27 @@ def _dataprojection(self, data):
         cut = tensorlib.shape(data)[0] - len(self.config.auxdata)
         return data[cut:]
 
+    def make_pdf(self, pars):
+        indices = []
+        pdfobjs = []
+
+        gaussian_pdf = self.constraints_gaussian.make_pdf(pars)
+        if gaussian_pdf:
+            indices.append(self.constraints_gaussian._normal_data)
+            pdfobjs.append(gaussian_pdf)
+
+        poisson_pdf = self.constraints_poisson.make_pdf(pars)
+        if poisson_pdf:
+            indices.append(self.constraints_poisson._poisson_data)
+            pdfobjs.append(poisson_pdf)
+
+        if pdfobjs:
+            simpdf = prob.Simultaneous(pdfobjs, indices)
+            return simpdf
+
     def logpdf(self, auxdata, pars):
-        tensorlib, _ = get_backend()
-        normal = self.constraints_gaussian.logpdf(auxdata, pars)
-        poisson = self.constraints_poisson.logpdf(auxdata, pars)
-        return prob.joint_logpdf([normal, poisson])
+        simpdf = self.make_pdf(pars)
+        return simpdf.log_prob(auxdata)
 
 
 class _MainModel(object):
```

Comment:
The class and the class methods should have docstrings.

Reply:
done

Reply:
done